### ISSUE TITLE
Check older development commits and reapply if needed

### ISSUE THREAD

**Issue by ferki** [RexOps/Rex#1265]
TL;DR: this is a placeholder issue to collect some of the older changes that got merged into the historical `development` branch, but didn't make it into mainline `master` after all, and see if they are useful to (re)apply now.

Longer version:

Around 1.4.1 release, development happened on dual tracks in `development` and `development-1.x` branches. Since `development` was the place for more experimental changes (as in "exploring ideas and preparing for v2"), `development-1.x` was used to do 1.x releases.

After a while, progress in `development` stalled and `development-1.x` remained active. Because of this, the latter became the new `master` when we consolidated the development workflow later.

This means there might be some changes in the old `development` branch which would be useful today, but haven't found their way into master yet. Let's play it safe, check the history and see which ones can be (re)applied.

**Comment by ferki** [comment]
FTR, I'm just attaching my initial notes on the review of the potentially affected range of commits.

[merge_backport_review.txt](https://github.com/RexOps/Rex/files/4678580/merge_backport_review.txt)

Legend for markers on the first character of the line:

- `x`: no further todo (done, all good, or reimplemented since then, not important, etc.)
- `!`: might need further investigation (usually because (partly) reimplemented, or mismerged as part of other commits)
- `-`: wasn't merged, need review/decision about merge as part of this issue

Comments are separated from commit messages with ` ==>` at the end of line.

NB.: pay attention that many commits are referencing another line in the same document since they were cherry-picked onto another branch and got a different commit ID due to that (sometimes even the content changed to reflect differences between the then-current `development` and `development-1.x` branches).
